aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/AWSPredictionsService.java

```diff
@@ -100,6 +105,36 @@ public void recognizeCelebrities(
         rekognitionService.recognizeCelebrities(image, onSuccess, onError);
     }
 
+    /**
+     * Delegate to {@link AWSRekognitionService} to detect plain text.
+     * @param image the Rekognition input image
+     * @param onSuccess triggered upon successful result
+     * @param onError triggered upon encountering error
+     */
+    public void detectPlainText(
+            @NonNull Image image,
```

Comment:
This method is weird in that it takes an AWS  service type as an input, but then renders an Amplify type as an output.

Or in other words, right now,`AWSPredictionsService` is "leaking" some abstractions.

Let's consider the duties of the `AWSPredictionsService`, vs the `AWSRecognitionService`.

The lower level ones like `AWSRecognitionService` are handling the business of creating service-modeled requests, and decoding service-modeled responses. But, they are still requiring some types like `Image`, `Document`, which are contained in the requests.

Theoretically, you could "patch" the "leak" even here. Your `AWSRekognitionService` could accept only Amplify types, and return only Amplify types.

Then, your `AWSPredictionsService` is just a bundle of the various service clients.

But certainly, _at minimum_ you should encapsulate the AWS services underneath the `AWSPredictionsService`.

I do believe that `AWSPredictionsService` is an architecturally desirable component. If you look at some other plugin modules, they `WhateverPlugin extends Plugin` are chokepoints, and get huge. And its because the developer(s) encoded business rules directly into the plugin classes, instead of encapsulating the behaviors into a self contained service component, as you have here. The `AWSPredictionsPlugin` is just a platform-binding adapter, once you have this service class.

Anyway, you're leaking the AWS stuff up through the service, into the plugin. The plugin would need to know what an `Image` is, and that's not so desirable.

Reply:
Reminder @raphkim to refactor this

Reply:
^ Or just the same `Bitmap` you got in the input, right? That was you don't have to do much mapping between the plugin and the service. Can you map from `Bitmap` (Android) to `Image` (AWS service model) inside this service class?

Reply:
I like leaving the `Bitmap` -> `ByteBuffer` conversion work inside `AWSImageIdentifyRequest` object, where it is actually converting the Amplify input into an acceptable service parameter. So I'll just pass the `ByteBuffer` directly to the services.